### ISSUE TITLE
QuickSight Describe/List DataSets

### ISSUE THREAD

**Issue by derrikf-spindance** [iann0036/former2#173]
I was having some trouble in getting QuickSight-related resources imported using the managed ReadOnlyAccess role. I came here and found #153 #163 which helped get a majority of everything. However Former2 is still not able to find the DataSets. I don't see anything related printing to the console and the policy appears to give the required access.

```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "VisualEditor0",
            "Effect": "Allow",
            "Action": [
                "quicksight:Describe*",
                "quicksight:List*"
            ],
            "Resource": "*"
        }
    ]
}
```

**Comment by derrikf-spindance** [comment]
@ianmckay-vha 

To clarify, after adding the policy above which I grabbed from one of the other related issue threads, the following changed:

I was able to successfully retrieve
- Analyses
- Dashboards
- DataSources

I was not able to successfully retrieve
- DataSets

Because my current project does not use the following, I was not able to verify
- Templates
- Themes
- Groups

**Comment by iann0036** [comment]
Thank for confirming.

I don't see anything obvious in the codebase for this, though I do know QuickSight is a little fiddly with permissions due to its nature. Will get back to you shortly after I attempt to reproduce myself.

**Comment by iann0036** [comment]
Hey @derrikf-spindance,

I've spotted a couple issues, one of which was an issue when getting the wrong property which is now fixed, so feel free to re-test now.

If that doesn't work, you might also be facing a different issue depending upon the dataset type, as the AWS APIs currently show the following error for some:

```
Error calling QuickSight.describeDataSet
DataSet type flatFile is not supported through API yet
```

For that, I'll have to go bother the relevant team to actually implement.

Hope that all helps!
